plugin.rb

```diff
@@ -56,6 +57,19 @@ class Engine < ::Rails::Engine
 
   reloadable_patch do
     CategoryListSerializer.class_eval { prepend GlobalFilter::CategoryListSerializerExtension }
+    CategoryDetailedSerializer.class_eval { prepend GlobalFilter::CategoryDetailedSerializerExtension }
+  end
+
+  add_to_serializer(:site, :filter_tags_total_topic_count) do
+    counts = {}
+    GlobalFilter::FilterTag.find_each do |gft|
+      counts[gft.name] = gft.total_topic_count
+    end
+    counts
+  end
+
+  add_to_serializer(:category_detailed, :filter_tag) do
+    scope.user&.custom_fields&.dig('global_filter_preference') || ""
```

Comment:
Do we need to also cover anonymous visitors here?

Reply:
Anon mode needs to be tweaked in multiple places. I will address a fix in a separate PR.